class.jetpack.php

```diff
@@ -3188,14 +3189,99 @@ public static function get_activation_source( $referer_url ) {
 	/**
 	 * Runs before bumping version numbers up to a new version
 	 *
-	 * @param  string $version    Version:timestamp
-	 * @param  string $old_version Old Version:timestamp or false if not set yet.
-	 * @return null              [description]
+	 * @param string $version    Version:timestamp.
+	 * @param string $old_version Old Version:timestamp or false if not set yet.
 	 */
 	public static function do_version_bump( $version, $old_version ) {
-		if ( ! $old_version ) { // For new sites
-			// There used to be stuff here, but this seems like it might  be useful to someone in the future...
+		if ( $old_version ) { // For existing Jetpack installations.
+			self::send_update_modal_data();
+		}
+	}
+
+	/**
+	 * Prepares the release post content and image data and saves it in the
+	 * state array. This data is used to create the update modal.
+	 */
+	public static function send_update_modal_data() {
+		$post_data = self::get_release_post_data();
+
+		if ( ! isset( $post_data ) || ! isset( $post_data['posts'] ) || ! isset( $post_data['posts'][0] ) ) {
+			return;
+		}
+
+		$post_content = isset( $post_data['posts'][0]['content'] ) ? $post_data['posts'][0]['content'] : null;
+		if ( empty( $post_content ) ) {
+			return;
+		}
+
+		$post_link = isset( $post_data['posts'][0]['URL'] ) ? $post_data['posts'][0]['URL'] : null;
+		if ( empty( $post_link ) ) {
+			return;
+		}
+
+		$post_image_src = isset( $post_data['posts'][0]['featured_image'] ) ? $post_data['posts'][0]['featured_image'] : null;
+
+		$content = wp_kses_post( $post_content );
+
+		// Remove the hidden elements from the modal content.
+		$dom_doc = new DOMDocument();
```

Comment:
Can we check if this library is installed before we used it?
Since some host do not have it installed by default.

Reply:
Good point - we should check before using it. I'll fix that.

Reply:
@enejb  - I added a check for the DOMDocument class in aaf22fc. If the class isn’t installed, we just bail and don’t show the update notice at all.

Do you know if it’s common for this library to not be installed on hosts? If it is, we could remove the hidden elements on the front-end instead.